src/core/src/service_interfaces/LifecycleManager.py

```diff
@@ -64,18 +64,29 @@ def read_extension_sequence(self):
                     self.composite_logger.log_error("Unable to read extension state file (retries exhausted). [Exception={0}]".format(repr(error)))
                     raise
 
-    def read_core_sequence(self):
-        """ Reads the core sequence file, but additionally establishes if this class is allowed to write to it when the freshest data is evaluated. """
-        self.composite_logger.log_debug("Reading core sequence...")
+    def identify_and_mitigate_core_sequence_issues(self):
+        """ Checks for issues with the core sequence file (file not exists, is dir, etc) and attempts to mitigate them. """
         if not os.path.exists(self.core_state_file_path) or not os.path.isfile(self.core_state_file_path):
             # Neutralizes directories
             if os.path.isdir(self.core_state_file_path):
                 self.composite_logger.log_error("Core state file path returned a directory. Attempting to reset.")
                 shutil.rmtree(self.core_state_file_path)
             # Writes a vanilla core sequence file
+            self.composite_logger.log_warning("Core state file did not exist. Attempting to reset.")
+            self.read_only_mode = False
+            self.update_core_sequence()
```

Comment:
Abstract into private method def write_default_core_sequence_file()
- repeated here and 79 to 81.